gnosis/cowswap
# [presign] Enable smart contracts

# Summary

This does the following:
* Adds a new flag for the `useWalletInfo` hook.
* This flag is the `allowsOffchainSigning` which can be used throughout the UI to change all the behaviours to be in "pre-sign" mode.
* Enables Smart Contract Wallets: So now you should be able to approve tokens (but not swap them, yet...)
* Removes some references to the `gassless nature` of cowswap


## If you connect with an EOA (i.e. Metamask)

You should see all references to gassless nature:


<img width="794" alt="Screenshot at Aug 26 16-18-46" src="https://user-images.githubusercontent.com/2352112/130981571-81f70a42-4c1a-4a37-b830-ef895510774c.png">


<img width="901" alt="Screenshot at Aug 26 16-19-15" src="https://user-images.githubusercontent.com/2352112/130981620-a74cc4d9-bb74-4852-8b31-d5969ca4197e.png">

## If you connect with an Smart contract (i.e. Gnosis Safe)
The references are gone

<img width="773" alt="Screenshot at Aug 26 16-20-08" src="https://user-images.githubusercontent.com/2352112/130981743-1fa7d789-ab40-48fc-829c-5eaf3d809a2b.png">



<img width="1140" alt="Screenshot at Aug 26 16-24-04" src="https://user-images.githubusercontent.com/2352112/130981841-0a5deda6-7664-44f6-b44b-118baa44878f.png">

# not included
* Final messages, or final UX
* Able to trade, u will get

<img width="1215" alt="Screenshot at Aug 26 16-24-27" src="https://user-images.githubusercontent.com/2352112/130981961-20d52788-1518-474e-9548-f98383ac4702.png">


# To Test

1. Try EOA. Verify that is like in production
2. Try Smart Contract. 
* Review if I left any important reference
* Try to approve a token
